compiler/rustc_middle/src/mir/interpret/allocation/provenance_map.rs

```diff
@@ -117,7 +132,7 @@ impl<Prov: Provenance> ProvenanceMap<Prov> {
     /// limit access to provenance outside of the `Allocation` abstraction.
     ///
     pub fn range_empty(&self, range: AllocRange, cx: &impl HasDataLayout) -> bool {
-        self.range_get_ptrs(range, cx).is_empty() && self.range_get_bytes(range).is_empty()
+        self.range_get_ptrs_is_empty(range, cx) && self.range_get_bytes(range).is_empty()
```

Comment:
We could do something similar for checking that the bytewise provenance is empty in this range, couldn't we?

Reply:
We could, but for the benchmarks `range_get_bytes` wasn't hot at all, while `range_get_ptrs` was very hot, so I didn't bother.

Reply:
Yeah the per-byte provenance is empty most of the time. Still it just seems oddly asymmetric now, making one wonder if there is a reason for the asymmetry.

Reply:
To be more precise, it is *always* empty for CTFE. It will only ever be non-empty in Miri.